src/Castle.Core/DynamicProxy/Internal/AttributeUtil.cs

```diff
@@ -243,13 +245,18 @@ private static bool ShouldSkipAttributeReplication(Type attribute)
 				return false;
 			}
 
-			var attrs = attribute.GetTypeInfo().GetCustomAttributes<AttributeUsageAttribute>(true).ToArray();
-			if (attrs.Length != 0)
+			if (!ignoreInheritance)
 			{
-				return attrs[0].Inherited;
+				var attrs = attribute.GetTypeInfo().GetCustomAttributes<AttributeUsageAttribute>(true).ToArray();
+				if (attrs.Length != 0)
+				{
+					return attrs[0].Inherited;
+				}
+
+				return true;
 			}
 
-			return true;
+			return false;
 		}
 
 		private static bool SpecialCaseAttributeThatShouldNotBeReplicated(Type attribute)
```

Comment:
While we are here cleaning things up could you inline this method, it just complicates things for no reason, there is only the one call site for `SpecialCaseAttributeThatShouldNotBeReplicated` anyway.

If you want to rebase anything nows the time and we'll get this one merged.

Reply:
I was thinking the same thing. Will do.